api-1.10/src/main/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,113 @@
+	    	}
+	    	else if (drugOrderCohortDefinition.getOnlyCurrentlyActive() == false) {
+		    	  query.whereNotNull("drugOrder.dateActivated").and().whereLess("drugOrder.dateStopped", new Date());
+	    	}
+	    }
+	    if (drugOrderCohortDefinition.getDrugSets() != null) {
+	    	
+	    	if (drugOrderCohortDefinition.getWhich() == "ANY") {
+		    	query.whereInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugSets().toArray());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich() == "ALL") {
+		    	query.whereIn("drugOrder.concept", drugOrderCohortDefinition.getDrugSets());
+		    	query.groupBy(
+			        		"drugOrder.patient.patientId" + " having count(distinct drugOrder.concept.conceptId) = " + drugOrderCohortDefinition.getDrugSets().size());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich() == "NONE") {
+		    	query.whereNotInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugSets());
+	    	}
+	    }
+	
+	    if (drugOrderCohortDefinition.getDrugConcepts() != null) {
+	    	if (drugOrderCohortDefinition.getWhich().equals("ANY")) {
+	    		query.whereInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugConcepts().toArray());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich().equals("ALL")) {
+	    		query.whereIn("drugOrder.concept", drugOrderCohortDefinition.getDrugConcepts());
+	    		query.groupBy(
+	    				"drugOrder.patient.patientId" + " having count(distinct drugOrder.concept.conceptId) = " + drugOrderCohortDefinition.getDrugSets().size());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich() == "NONE") {
+	    		query.whereNotInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugConcepts());
+	    	}
+	    }
+	
+	    if (drugOrderCohortDefinition.getActivatedOnOrBefore() != null) {
+	    	query.whereLessOrEqualTo("drugOrder.dateActivated", drugOrderCohortDefinition.getActivatedOnOrBefore());
+	    }
+	    if (drugOrderCohortDefinition.getActivatedOnOrAfter() != null) {
+	    	query.whereGreaterOrEqualTo("drugOrder.dateActivated", drugOrderCohortDefinition.getActivatedOnOrAfter());
+	    }
+	
+	    if (drugOrderCohortDefinition.getCareSetting() != null) {
+	    	query.whereEqual("drugOrder.careSetting", drugOrderCohortDefinition.getCareSetting());
+	    }
+	
+	    if (drugOrderCohortDefinition.getActiveWithinLastDays() != null) {
+	    	Integer days = drugOrderCohortDefinition.getActiveWithinLastDays();
+	    	query.whereNotNull("drugOrder.dateActivated").and()
+	    		 .whereGreaterOrNull("drugOrder.dateStopped", DateUtils.addDays(new Date(), -days));
+	    }
+	
+	    if (drugOrderCohortDefinition.getActiveWithinLastMonths() != null) {
+	    	Integer months = drugOrderCohortDefinition.getActiveWithinLastMonths();
+	    	query.whereNotNull("drugOrder.dateActivated").and()
+	    		 .whereGreaterOrNull("drugOrder.dateStopped", DateUtils.addMonths(new Date(), -months));
+	    }
```

Comment:
We would need to change the evaluator to reflect the property changes that I am suggesting.

Reply:
I have resolved this.